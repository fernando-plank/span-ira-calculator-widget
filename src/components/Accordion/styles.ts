import styled from 'styled-components'

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: baseline & > svg {
    cursor: pointer;
  }
`

export const Body = styled.div`
  display: flex;
  height: 100%;
  opacity: 0;
  max-height: 0;
  pointer-events: none;
  margin-bottom: 17px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    flex-direction: column;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: row;
  }

  &.active {
    max-height: max-content;
    opacity: 1;
    pointer-events: auto;
  }
`

export const Title = styled.h2`
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  width: 80%;
  margin-bottom: 17px;
`

export const Separator = styled.div`
  height: 1px;
  background: #b0b0b1;
  border-radius: 2px;
  margin-bottom: 17px;
`
