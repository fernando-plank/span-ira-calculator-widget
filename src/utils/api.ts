import { IncentivesResponse, TaxCreditsIncentives } from '@services/types'
import { makeTaxCreditsTooltip } from './tooltips'

export const formatTaxCreditsIncentives = (
  tax_credit_incentives: TaxCreditsIncentives[]
): TaxCreditsIncentives[] => {
  let eletricPanel: any = null
  let items: TaxCreditsIncentives[] = tax_credit_incentives.filter((item) => {
    if (item.item === 'Electric Panel') {
      eletricPanel = item
    }
    return item.type === 'tax_credit' && item.item !== 'Electric Panel'
  })

  items = items?.filter(
    (item, index, self) =>
      index ===
      self.findIndex((t) => {
        return t.item === item.item
      })
  )

  if (items && eletricPanel) {
    eletricPanel.amount = 600
    eletricPanel['item'] = 'SPAN Smart Electrical Panel'
    items.unshift(eletricPanel)
    items = makeTaxCreditsTooltip(items)
  }

  return items
}
