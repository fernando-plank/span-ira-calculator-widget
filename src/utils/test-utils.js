var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
