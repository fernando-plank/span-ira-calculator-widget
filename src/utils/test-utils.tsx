/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
