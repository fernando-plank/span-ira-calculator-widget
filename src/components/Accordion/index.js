import React, { useState } from 'react';
import Icon from '@components/Icon';
import classname from 'classnames';
import * as S from './styles';
const Accordion = (props) => {
    const [showSection, setShowSection] = useState(false);
    return (React.createElement(React.Fragment, null,
        React.createElement(S.Header, null,
            React.createElement(S.Title, { "data-testid": "accordion-title", style: props.style, onClick: () => setShowSection(!showSection) }, props.title),
            React.createElement(Icon, { icon: props.icon, onClick: () => setShowSection(!showSection) })),
        React.createElement(S.Separator, null),
        React.createElement(S.Body, { "data-testid": "accordion-body", className: classname({ active: showSection }) }, props.children)));
};
export default Accordion;
