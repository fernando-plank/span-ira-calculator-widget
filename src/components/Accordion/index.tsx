import React, { useState } from 'react'

import Icon from '@components/Icon'
import classname from 'classnames'

import * as S from './styles'

export interface AccordionProps {
  children: React.ReactNode
  icon: string
  title: string
  style?: object
}

const Accordion = (props: AccordionProps) => {
  const [showSection, setShowSection] = useState(false)
  return (
    <>
      <S.Header>
        <S.Title
          data-testid="accordion-title"
          style={props.style}
          onClick={() => setShowSection(!showSection)}
        >
          {props.title}
        </S.Title>
        <Icon icon={props.icon} onClick={() => setShowSection(!showSection)} />
      </S.Header>
      <S.Separator></S.Separator>
      <S.Body
        data-testid="accordion-body"
        className={classname({ active: showSection }, 'accordion-body')}
      >
        {props.children}
      </S.Body>
    </>
  )
}

export default Accordion
