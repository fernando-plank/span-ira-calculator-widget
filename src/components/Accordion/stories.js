import React from 'react';
import Accordion from '.';
export default {
    title: 'Accordion',
    component: Accordion
};
export const Default = () => (React.createElement(Accordion, { icon: "Plus", title: "Title" }, "Description"));
