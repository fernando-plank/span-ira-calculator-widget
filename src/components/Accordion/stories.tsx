import React from 'react'
import { Story, Meta } from '@storybook/react'

import Accordion from '.'

export default {
  title: 'Accordion',
  component: Accordion
} as Meta

export const Default: Story = () => (
  <Accordion icon="Plus" title="Title">
    Description
  </Accordion>
)
