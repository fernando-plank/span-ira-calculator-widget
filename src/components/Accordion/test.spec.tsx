import { fireEvent, screen } from '@testing-library/react'
import { render } from '../../utils/test-utils'

import Accordion from '.'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<Accordion />', () => {
  it('should add title', () => {
    const { container } = render(
      <Accordion
        style={{ backgroundColor: 'red' }}
        icon={'Plus'}
        title={'Title'}
      >
        Some text
      </Accordion>
    )

    expect(screen.getByText('Title')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should add classname on click', () => {
    const { container } = render(
      <Accordion icon={'Plus'} title={'Title'}>
        Some text
      </Accordion>
    )

    fireEvent.click(screen.getByTestId('accordion-title'))

    expect(screen.getByTestId('accordion-body')).toHaveClass('active')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render accordion body', () => {
    const { container } = render(
      <Accordion icon={'Plus'} title={'Title'}>
        Some text
      </Accordion>
    )

    expect(screen.getByText('Some text')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
