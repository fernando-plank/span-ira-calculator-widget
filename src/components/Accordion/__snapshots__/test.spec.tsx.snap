// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should add classname on click 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

align-items:baseline .c0 > svg {
  cursor: pointer;
}

.c3 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c5 {
  width: 2px;
  height: 24px;
  background-color: #1f1f1f;
  border-radius: 2px;
  position: absolute;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c5.active {
  -webkit-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -ms-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
}

.c4 {
  width: 24px;
  height: 2px;
  background-color: #1f1f1f;
  border-radius: 2px;
}

.c2 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  width: 100%;
}

@media (min-width:320px) {

}

@media (min-width:1024px) {

}

<div
  class="c0 c1"
>
  <h2
    class="c2"
    data-testid="accordion-title"
  >
    Title
  </h2>
  <div
    class="c3"
  >
    <div
      class="c4"
    />
    <div
      class="c5 active accordion-body"
    />
  </div>
</div>
`;

exports[`<Accordion /> should add title 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

align-items:baseline .c0 > svg {
  cursor: pointer;
}

.c3 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c5 {
  width: 2px;
  height: 24px;
  background-color: #1f1f1f;
  border-radius: 2px;
  position: absolute;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c5.active {
  -webkit-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -ms-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
}

.c4 {
  width: 24px;
  height: 2px;
  background-color: #1f1f1f;
  border-radius: 2px;
}

.c2 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  width: 100%;
}

@media (min-width:320px) {

}

@media (min-width:1024px) {

}

<div
  class="c0 c1"
>
  <h2
    class="c2"
    data-testid="accordion-title"
    style="background-color: red;"
  >
    Title
  </h2>
  <div
    class="c3"
  >
    <div
      class="c4"
    />
    <div
      class="c5 accordion-body"
    />
  </div>
</div>
`;

exports[`<Accordion /> should render accordion body 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

align-items:baseline .c0 > svg {
  cursor: pointer;
}

.c3 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c5 {
  width: 2px;
  height: 24px;
  background-color: #1f1f1f;
  border-radius: 2px;
  position: absolute;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c5.active {
  -webkit-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -ms-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
}

.c4 {
  width: 24px;
  height: 2px;
  background-color: #1f1f1f;
  border-radius: 2px;
}

.c2 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  width: 100%;
}

@media (min-width:320px) {

}

@media (min-width:1024px) {

}

<div
  class="c0 c1"
>
  <h2
    class="c2"
    data-testid="accordion-title"
  >
    Title
  </h2>
  <div
    class="c3"
  >
    <div
      class="c4"
    />
    <div
      class="c5 accordion-body"
    />
  </div>
</div>
`;
